libp2p/peer/peerdata.py

```diff
@@ -1,35 +1,48 @@
+from typing import (
+    Any,
+    Dict,
+    List,
+    Sequence,
+)
+
+from multiaddr import Multiaddr
+
 from .peerdata_interface import IPeerData
 
 
 class PeerData(IPeerData):
 
-    def __init__(self):
+    metadata: Dict[Any, Any]
+    protocols: List[str]
+    addrs: List[Multiaddr]
+
+    def __init__(self) -> None:
         self.metadata = {}
         self.protocols = []
         self.addrs = []
 
-    def get_protocols(self):
+    def get_protocols(self) -> List[str]:
         return self.protocols
 
-    def add_protocols(self, protocols):
-        self.protocols.extend(protocols)
+    def add_protocols(self, protocols: Sequence[str]) -> None:
+        self.protocols.extend(list(protocols))
 
-    def set_protocols(self, protocols):
-        self.protocols = protocols
+    def set_protocols(self, protocols: Sequence[str]) -> None:
+        self.protocols = list(protocols)
 
-    def add_addrs(self, addrs):
-        self.addrs.extend(addrs)
+    def add_addrs(self, addrs: Sequence[Multiaddr]) -> None:
+        self.addrs.extend(list(addrs))
```

Comment:
`extend(Iterable)` should already be able to consume `Sequence` which is also a subtype of `Iterable`, without `list`.
Ref: https://docs.python.org/3/library/typing.html#typing.Collection